src/antelope/wallets/authenticators/ExternalProviderAuth.ts

```diff
@@ -0,0 +1,138 @@
+            value,
+        }).then(
+            (transaction: ethers.providers.TransactionResponse) => transaction,
+        ).catch((error) => {
+            if ('ACTION_REJECTED' === ((error as {code:string}).code)) {
+                throw new AntelopeError('antelope.evm.error_transaction_canceled');
+            } else {
+                // unknown error we print on console
+                console.error(error);
+                throw new AntelopeError('antelope.evm.error_send_transaction', { error });
+            }
+        });
+    }
+
+    // EVMAuthenticator API ----------------------------------------------------------
+
+    async login(network: string): Promise<addressString | null> {
+        this.trace('login', network);
+        useFeedbackStore().setLoading(`${this.getName()}.login`);
+        const response = await super.login(network);
+        useFeedbackStore().unsetLoading(`${this.getName()}.login`);
+        return response;
+    }
+
+    async logout(): Promise<void> {
+        this.trace('logout');
+    }
+
+    async getSystemTokenBalance(address: addressString | string): Promise<ethers.BigNumber> {
+        this.trace('getSystemTokenBalance', address);
+        const provider = await this.web3Provider();
+        if (provider) {
+            return provider.getBalance(address);
+        } else {
+            throw new AntelopeError('antelope.evm.error_no_provider');
+        }
+    }
+
+    async getERC20TokenBalance(account: addressString | string, tokenAddress: addressString | string): Promise<ethers.BigNumber> {
+        this.trace('getERC20TokenBalance', [account, tokenAddress]);
+        const erc20ABI = useEVMStore().getTokenABI(ERC20_TYPE) as AbiItem[];
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const web3 = new Web3(this.getProvider() as any);
+        const contract = new web3.eth.Contract(erc20ABI, tokenAddress);
+        return contract.methods.balanceOf(account).call()
+            .then((balance: never) => ethers.BigNumber.from(balance));
+    }
+
+    async transferTokens(token: TokenClass, amount: ethers.BigNumber, to: addressString): Promise<EvmTransactionResponse> {
+        this.trace('transferTokens', token, amount, to);
+        if (token.isSystem) {
+            return this.sendSystemToken(to, amount);
+        } else {
+            const evm = useEVMStore();
+            const contract = await evm.getContract(this, token.address, token.type);
+            if (contract) {
+                const contractInstance = await contract.getContractInstance();
+                const amountInWei = amount.toString();
+                return contractInstance.transfer(to, amountInWei);
+            } else {
+                throw new AntelopeError('antelope.balances.error_token_contract_not_found', { address: token.address });
+            }
+        }
+    }
+
+    prepareTokenForTransfer(token: TokenClass | null, amount: ethers.BigNumber, to: string): Promise<void> {
```

Comment:
can you please explain the purpose of this method to me?

Reply:
This is the origin of the problem that @donnyquixotic found and fixed it. It is WalletConnect/iOS-specific issue and the prepareTokenForTransfer function is part of the solution.

https://github.com/telosnetwork/telos-wallet/blob/develop/src/pages/evm/wallet/SendPage.vue#L224-L247